docs/rules/no-unnecessary-index-route.md

```diff
@@ -0,0 +1,46 @@
+## Disallow unnecessary `index` route definition
+
+### Rule name: `no-unnecessary-index-route`
+
+The `index` route (for the `/` path) is automatically provided at every level of nesting and generally does not need to be defined in the router.
+
+The only allowed use cases for the `index` route definition are:
+
+* Dynamic segments
+* Wildcards
```

Comment:
I'm wondering why we allow these special cases. Are those snippets below actually useful? Since `index` has a special meaning I'm not sure if we should even allow a `path` option for an `index` route definition. Am I missing something? 🤔

Reply:
I agree that it would not be a best practice to use the `index` route for those special cases. So I have removed the special cases from this PR. The good news is that this simplifies the new rule a lot.